src/main/java/cubicchunks/worldgen/generator/custom/CustomTerrainGenerator.java

```diff
@@ -76,8 +76,8 @@
 @MethodsReturnNonnullByDefault
 public class CustomTerrainGenerator extends BasicCubeGenerator {
 
-    private static final int CACHE_SIZE_2D = 16 * 16;
-    private static final int CACHE_SIZE_3D = 16 * 16 * 16;
+    private static final int CACHE_SIZE_2D = Cube.SIZE * Cube.SIZE;
+    private static final int CACHE_SIZE_3D = Cube.SIZE * Cube.SIZE * Cube.SIZE;
```

Comment:
This 16 isn't directly related to cube size. The number is mostly a coincidence.

If you really wanted to use constants here, it would need to be something like 
```java
closestPowerOf2BiggerThan((Cube.SIZE / min(GENERATION_SECTION_SIZE_XZ, GENERATION_SECTION_SIZE_Y) + 1) * 3)
```
with `GENERATION_SECTION_SIZE_XZ=4` and `GENERATION_SECTION_SIZE_Y=8`

And the 5 below replaced with `(Cube.SIZE / min(GENERATION_SECTION_SIZE_XZ, GENERATION_SECTION_SIZE_Y) + 1)`

So this probably should be kept as is. Besides, the code will work fine anyway no matter the value, just slower or faster.